test/metabase/test/test_data.clj

```diff
@@ -0,0 +1,205 @@
+        (loop [acc init]
+          (if (reduced? acc)
+            (unreduced acc)
+            (if-let [object (edn/read {:eof nil, :readers {'t u.date/parse}} r)]
+              (recur (rf acc object))
+              acc)))))))
+
+(mu/defn ^:private read-edn-file
+  [filename :- :string]
+  (into [] (reducible-edn-file filename)))
+
+(mu/defn ^:private database-schema :- ::schema.database
+  []
+  (first (read-edn-file "test_resources/test_data/schema.edn")))
+
+(mu/defn ^:private table-schema :- ::schema.table
+  [table-name :- :string]
+  (m/find-first #(= (:name %) table-name)
+                (:tables (database-schema))))
+
+(mu/defn ^:private field-schema :- ::schema.field
+  [table-name :- :string
+   field-name :- :string]
+  (m/find-first #(= (:name %) field-name)
+                (:fields (table-schema table-name))))
+
+(defn schema
+  ([]                      (database-schema))
+  ([table-name]            (table-schema table-name))
+  ([table-name field-name] (field-schema table-name field-name)))
+
+(mu/defn ^:private table-rows-filename :- :string
+  [table-name :- ::lib.schema.common/non-blank-string]
+  (.getAbsolutePath (io/file (format "test_resources/test_data/rows_%s.edn" (munge table-name)))))
+
+(mu/defn ^:private reducible-table-rows :- ::reducible-rows
+  [table-name :- ::lib.schema.common/non-blank-string]
+  (reducible-edn-file (table-rows-filename table-name)))
+
+(mu/defn ^:private artifact-directory :- :string
+  [driver :- ::driver]
+  (.getAbsolutePath (io/file (format "target/test_data/artifacts/%s" (munge (u/qualified-name driver))))))
+
+(defmulti create-schema-artifacts-method!
+  {:arglists '([driver target-directory database-schema]), :added "0.51.0"}
+  tx/dispatch-on-driver-with-test-extensions
+  :hierarchy #'driver/hierarchy)
+
+(mu/defn create-schema-artifacts! :- ::artifacts
+  ([driver]
+   (create-schema-artifacts! driver (schema)))
+
+  ([driver schema]
+   (create-schema-artifacts! driver (artifact-directory driver) schema))
+
+  ([driver            :- ::driver
+    target-directory  :- :string
+    schema            :- ::schema.database]
+   (create-schema-artifacts-method! driver target-directory schema)))
+
+(defmulti create-data-artifacts-method!
+  {:arglists '([driver target-directory table-schema rows]), :added "0.51.0"}
+  tx/dispatch-on-driver-with-test-extensions
+  :hierarchy #'driver/hierarchy)
+
+(mu/defn create-data-artifacts! :- ::artifacts
+  ([driver table]
+   (create-data-artifacts! driver (artifact-directory driver) table))
+
+  ([driver target-directory table]
+   (create-data-artifacts! driver target-directory (reducible-table-rows (:name table))))
+
+  ([driver           :- ::driver
+    target-directory :- :string
+    table            :- ::schema.table
+    rows             :- ::reducible-rows]
+   (create-data-artifacts-method! driver target-directory table rows)))
```

Comment:
Extra arities of stuff for convenience from the REPL to make it easy to do something like 

```clj
(create-data-artifacts! :postgres (schema "venues"))
```

so you can inspect the output